### PR TITLE
[YARN-7772] - yarn.nodemanager.container-monitor.procfs-tree.smaps-based-rss.enabled is not taken if process-tree class is not set

### PR DESCRIPTION
**Pull request by ashangit**
Currently setConf is not call which leads to parameter
yarn.nodemanager.container-monitor.procfs-tree.smaps-based-rss.enabled not beaing read

### CODE REVIEW

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/util/TestResourceCalculatorProcessTree.java [hunk 1]
```diff
@@ -88,5 +88,5 @@ public void testCreatedInstanceConfigured() {
     tree = ResourceCalculatorProcessTree.getResourceCalculatorProcessTree("1", EmptyProcessTree.class, conf);
     assertNotNull(tree);
     assertThat(tree.getConf(), sameInstance(conf));
-  } 
+  }
```
Comment (Willymontaz) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/util/TestResourceCalculatorProcessTree.java]: OPT: Unnecessary change

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ResourceCalculatorProcessTree.java [hunk 2]
```diff
@@ -226,15 +236,9 @@ public static ResourceCalculatorProcessTree getResourceCalculatorProcessTree(
...
-      return new ProcfsBasedProcessTree(pid);
-    }
-    if (WindowsBasedProcessTree.isAvailable()) {
-      return new WindowsBasedProcessTree(pid);
-    }
-
     // Not supported on this system.
     return null;
   }
+
```
Comment (Willymontaz) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ResourceCalculatorProcessTree.java]: OPT: Unnecessary change

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ResourceCalculatorProcessTree.java [hunk 3]
```diff
@@ -213,7 +213,17 @@ public float getCpuUsagePercent() {
    *         is not available for this system.
    */
   public static ResourceCalculatorProcessTree getResourceCalculatorProcessTree(
-    String pid, Class<? extends ResourceCalculatorProcessTree> clazz, Configuration conf) {
+          String pid, Class<? extends ResourceCalculatorProcessTree> clazz, Configuration conf) {
```
Comment (benallamar) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ResourceCalculatorProcessTree.java]: it’s better to be kept intact

**Comment by ashangit** [comment]
Linked to https://issues.apache.org/jira/browse/YARN-7772

**Comment by Willymontaz** [comment]
LGTM
